app/models/merchant.rb

```diff
@@ -16,44 +16,34 @@ def self.most_items(quantity)
       .limit(quantity)
   end
 
-  def self.order_by_revenue(top_n)
-    result = []
-    all_revenues_per_merchant = total_revenues
-    top_n_revenues = all_revenues_per_merchant.values.sort.reverse.shift(top_n)
-    top_n_revenues.each do |revenue|
-      result << all_revenues_per_merchant.key(revenue)
-    end
-    result
-  end
-
-  def self.total_revenues
-    total_revenues = {}
-    all.each do |merchant|
-      total_revenues[merchant] = merchant.total_revenue.round(2)
-    end
-    total_revenues
```

Comment:
**Refactored these, as well as the methods below, to make all rspec and spec harness tests pass, and to use less Ruby.**